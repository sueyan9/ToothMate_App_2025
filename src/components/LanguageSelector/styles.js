import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  languageButton: {
    position: 'absolute',
    right: 24,
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#FFFFFF',
    borderRadius: 20,
    paddingHorizontal: 12,
    paddingVertical: 8,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
    zIndex: 2000,
  },
  languageButtonText: {
    fontSize: 16,
  },
  modalContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
  },
  modalContent: {
    backgroundColor: '#FFFFFF',
    borderRadius: 20,
    padding: 24,
    margin: 20,
    minWidth: 280,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.25,
    shadowRadius: 8,
    elevation: 5,
  },
  modalTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#333333',
    textAlign: 'center',
    marginBottom: 20,
  },
  languageOption: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 12,
    paddingHorizontal: 16,
    borderRadius: 12,
    marginVertical: 4,
  },
  selectedLanguage: {
    backgroundColor: '#E9F1F8',
    borderColor: '#516287',
    borderWidth: 1,
  },
  languageFlag: {
    fontSize: 24,
    marginRight: 12,
  },
  languageText: {
    fontSize: 16,
    color: '#333333',
  },
  selectedLanguageText: {
    color: '#516287',
    fontWeight: 'bold',
  },
  closeButton: {
    backgroundColor: '#516287',
    borderRadius: 12,
    paddingVertical: 12,
    marginTop: 16,
  },
  closeButtonText: {
    color: '#FFFFFF',
    fontSize: 16,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
