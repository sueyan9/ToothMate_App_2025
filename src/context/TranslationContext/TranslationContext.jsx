import { DEEPL_API_KEY } from '@env';
import AsyncStorage from '@react-native-async-storage/async-storage';
import axios from 'axios';
import createDataContext from '../createDataContext';

const translationReducer = (state, action) => {
  switch (action.type) {
    case 'set_language':
      return { ...state, currentLanguage: action.payload };
    case 'set_translations':
      return { 
        ...state, 
        translations: { 
          ...state.translations, 
          [action.payload.language]: action.payload.translations 
        } 
      };
    case 'set_loading':
      return { ...state, isLoading: action.payload };
    case 'set_error':
      return { ...state, error: action.payload };
    default:
      return state;
  }
};

const setLanguage = (dispatch) => async (language) => {
  try {
    await AsyncStorage.setItem('selectedLanguage', language);
    dispatch({ type: 'set_language', payload: language });
  } catch (error) {
    console.error('Error saving language preference:', error);
  }
};

const translateText = (dispatch) => async (texts, targetLanguage) => {
  if (targetLanguage === 'en') {
    // No translation needed for English
    return texts;
  }

  try {
  //  dispatch({ type: 'set_loading', payload: true });
    dispatch({ type: 'set_error', payload: null });

    // Convert language code to DeepL format
    let deeplLangCode = targetLanguage.toUpperCase();
    if (targetLanguage === 'es') {
      deeplLangCode = 'ES'; // Spanish
    } else if (targetLanguage === 'zh') {
      deeplLangCode = 'ZH'; // Chinese (simplified)
    } else if (targetLanguage === 'nl') {
      deeplLangCode = 'NL'; // Dutch
    } else if (targetLanguage === 'mi') {
      // Maori is not supported by DeepL, return original texts
      console.log('Maori language not supported by DeepL API, returning original texts');
      return texts;
    }

    console.log('Translating to:', deeplLangCode);
    console.log('API Key exists:', !!DEEPL_API_KEY);
    console.log('Texts to translate:', texts);

    // Create form data for the request
    const params = new URLSearchParams();
    
    // Filter out empty or invalid texts
    const validTexts = texts.filter(text => text && text.trim().length > 0);
    if (validTexts.length === 0) {
      console.warn('No valid texts to translate');
      return texts;
    }
    
    validTexts.forEach(text => {
      params.append('text', text.trim());
    });
    params.append('target_lang', deeplLangCode);
    params.append('source_lang', 'EN');

    const response = await axios.post(
      'https://api-free.deepl.com/v2/translate',
      params.toString(),
      {
        headers: {
          'Authorization': `DeepL-Auth-Key ${DEEPL_API_KEY}`,
          'Content-Type': 'application/x-www-form-urlencoded'
        },
        timeout: 10000 // 10 second timeout
      }
    );

    console.log('Translation response:', response.data);

    const translations = response.data.translations.map(t => t.text);
    
    dispatch({ 
      type: 'set_translations', 
      payload: { 
        language: targetLanguage, 
        translations: texts.reduce((acc, text, index) => {
          acc[text] = translations[index];
          return acc;
        }, {})
      }
    });

    return translations;
  } catch (error) {
    console.error('Translation error details:', {
      message: error.message,
      status: error.response?.status,
      data: error.response?.data,
      headers: error.response?.headers
    });
    
    // Provide more specific error messages
    let errorMessage = error.message;
    if (error.response?.status === 400) {
      errorMessage = 'Translation request failed: Invalid parameters or unsupported language';
    } else if (error.response?.status === 403) {
      errorMessage = 'Translation failed: API key is invalid or expired';
    } else if (error.response?.status === 456) {
      errorMessage = 'Translation failed: Usage limit reached';
    }
    
    dispatch({ type: 'set_error', payload: errorMessage });
    return texts; // Return original texts if translation fails
  } finally {
   // dispatch({ type: 'set_loading', payload: false });
  }
};

const loadLanguagePreference = (dispatch) => async () => {
  try {
    const savedLanguage = await AsyncStorage.getItem('selectedLanguage');
    if (savedLanguage) {
      dispatch({ type: 'set_language', payload: savedLanguage });
    }
  } catch (error) {
    console.error('Error loading language preference:', error);
  }
};

export const { Context, Provider } = createDataContext(
  translationReducer,
  { setLanguage, translateText, loadLanguagePreference },
  {
    currentLanguage: 'en', // Default to English
    translations: {},
    isLoading: false,
    error: null
  }
);
