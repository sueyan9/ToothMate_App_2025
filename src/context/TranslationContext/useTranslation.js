import { useContext, useEffect, useState } from 'react';
import { Context as TranslationContext } from './TranslationContext';

// Language mappings for DeepL API
const LANGUAGE_CODES = {
  'English': 'en',
  'Spanish': 'es',
  'Chinese': 'zh'
};

const LANGUAGE_DISPLAY_NAMES = {
  'en': 'English',
  'es': 'Spanish', 
  'zh': 'Chinese'
};

export const useTranslation = () => {
  const { 
    state: { currentLanguage, translations, isLoading, error }, 
    translateText, 
    setLanguage, 
    loadLanguagePreference 
  } = useContext(TranslationContext);

  const [translatedTexts, setTranslatedTexts] = useState({});

  useEffect(() => {
    loadLanguagePreference();
  }, []);

  // Clear local translations when language changes
  useEffect(() => {
    setTranslatedTexts({});
  }, [currentLanguage]);

  const t = (text) => {
    if (currentLanguage === 'en') {
      return text;
    }
    
    // Check context translations first, then local cache
    return translations[currentLanguage]?.[text] || translatedTexts[text] || text;
  };

  const translateAndCache = async (texts) => {
    if (currentLanguage === 'en') {
      return;
    }

    try {
      const translatedResults = await translateText(texts, currentLanguage);
      const newTranslations = {};
      texts.forEach((text, index) => {
        newTranslations[text] = translatedResults[index];
      });
      
      // Update local state immediately for faster UI updates
      setTranslatedTexts(prev => ({ ...prev, ...newTranslations }));
    } catch (error) {
      console.error('Translation failed:', error);
    }
  };

  const changeLanguage = async (language) => {
    await setLanguage(language);
  };

  const getAvailableLanguages = () => {
    return Object.keys(LANGUAGE_CODES);
  };

  const getCurrentLanguageDisplay = () => {
    return LANGUAGE_DISPLAY_NAMES[currentLanguage] || 'English';
  };

  return {
    t,
    currentLanguage,
    changeLanguage,
    translateAndCache,
    getAvailableLanguages,
    getCurrentLanguageDisplay,
    isLoading,
    error,
    LANGUAGE_CODES
  };
};
