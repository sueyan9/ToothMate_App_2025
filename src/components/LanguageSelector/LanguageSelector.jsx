import { useState } from 'react';
import { Modal, Text, TouchableOpacity, View } from 'react-native';
import { useTranslation } from '../../context/TranslationContext/useTranslation';
import styles from './styles';

const LanguageSelector = () => {
  const { 
    changeLanguage, 
    getAvailableLanguages, 
    getCurrentLanguageDisplay,
    LANGUAGE_CODES 
  } = useTranslation();
  
  const [modalVisible, setModalVisible] = useState(false);

  const handleLanguageChange = async (languageName) => {
    const languageCode = LANGUAGE_CODES[languageName];
    await changeLanguage(languageCode);
    setModalVisible(false);
  };

  const getLanguageFlag = (language) => {
    const flags = {
      'English': '🇺🇸',
      'Spanish': '🇪🇸',
      'Chinese': '🇨🇳',
      'Dutch': '🇳🇱'
    };
    return flags[language] || '🌐';
  };

  return (
    <>
      <TouchableOpacity 
        style={styles.languageButton} 
        onPress={() => setModalVisible(true)}
      >
        <Text style={styles.languageButtonText}>
          🌐 {getLanguageFlag(getCurrentLanguageDisplay())}
        </Text>
      </TouchableOpacity>

      <Modal
        animationType="slide"
        transparent={true}
        visible={modalVisible}
        onRequestClose={() => setModalVisible(false)}
      >
        <View style={styles.modalContainer}>
          <View style={styles.modalContent}>
            <Text style={styles.modalTitle}>Select Language</Text>
            
            {(getAvailableLanguages() || []).map((language) => (
              <TouchableOpacity
                key={language}
                style={[
                  styles.languageOption,
                  getCurrentLanguageDisplay() === language && styles.selectedLanguage
                ]}
                onPress={() => handleLanguageChange(language)}
              >
                <Text style={styles.languageFlag}>
                  {getLanguageFlag(language)}
                </Text>
                <Text style={[
                  styles.languageText,
                  getCurrentLanguageDisplay() === language && styles.selectedLanguageText
                ]}>
                  {language}
                </Text>
              </TouchableOpacity>
            ))}
            
            <TouchableOpacity
              style={styles.closeButton}
              onPress={() => setModalVisible(false)}
            >
              <Text style={styles.closeButtonText}>Close</Text>
            </TouchableOpacity>
          </View>
        </View>
      </Modal>
    </>
  );
};

export default LanguageSelector;
