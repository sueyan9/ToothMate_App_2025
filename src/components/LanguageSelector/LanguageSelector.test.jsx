import { fireEvent, render } from '@testing-library/react-native';
import { useTranslation } from '../../context/TranslationContext/useTranslation';
import LanguageSelector from './LanguageSelector';

// Mock the translation hook
jest.mock('../../context/TranslationContext/useTranslation', () => ({
  useTranslation: jest.fn(),
}));

// Mock styles
jest.mock('./styles', () => ({
  languageButton: {
    padding: 10,
    backgroundColor: '#f0f0f0',
    borderRadius: 5,
    marginHorizontal: 10,
  },
  languageButtonText: {
    fontSize: 16,
    textAlign: 'center',
  },
  modalContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.5)',
  },
  modalContent: {
    backgroundColor: 'white',
    padding: 20,
    borderRadius: 10,
    width: '80%',
    maxWidth: 300,
  },
  modalTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 15,
    textAlign: 'center',
  },
  languageOption: {
    padding: 15,
    borderBottomWidth: 1,
    borderBottomColor: '#eee',
  },
  languageOptionText: {
    fontSize: 16,
  },
  selectedLanguage: {
    backgroundColor: '#e3f2fd',
  },
  closeButton: {
    marginTop: 15,
    padding: 10,
    backgroundColor: '#007AFF',
    borderRadius: 5,
  },
  closeButtonText: {
    color: 'white',
    textAlign: 'center',
    fontSize: 16,
  },
}));

describe('LanguageSelector Snapshot Tests', () => {
  const mockTranslationHook = {
    changeLanguage: jest.fn(),
    getAvailableLanguages: jest.fn(),
    getCurrentLanguageDisplay: jest.fn(),
    LANGUAGE_CODES: {
      'English': 'en',
      'Spanish': 'es',
      'Chinese': 'zh',
      'Dutch': 'nl'
    }
  };

  beforeEach(() => {
    jest.clearAllMocks();
    useTranslation.mockReturnValue(mockTranslationHook);
  });

  describe('Closed Modal States', () => {
    it('should match snapshot with English selected', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue('English');
      mockTranslationHook.getAvailableLanguages.mockReturnValue(['English', 'Spanish', 'Chinese', 'Dutch']);

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot with Spanish selected', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue('Spanish');
      mockTranslationHook.getAvailableLanguages.mockReturnValue(['English', 'Spanish', 'Chinese', 'Dutch']);

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot with Chinese selected', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue('Chinese');
      mockTranslationHook.getAvailableLanguages.mockReturnValue(['English', 'Spanish', 'Chinese', 'Dutch']);

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot with Dutch selected', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue('Dutch');
      mockTranslationHook.getAvailableLanguages.mockReturnValue(['English', 'Spanish', 'Chinese', 'Dutch']);

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot with unknown language (fallback)', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue('French');
      mockTranslationHook.getAvailableLanguages.mockReturnValue(['English', 'Spanish', 'Chinese', 'Dutch']);

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Open Modal States', () => {
    it('should match snapshot with modal open - English current', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue('English');
      mockTranslationHook.getAvailableLanguages.mockReturnValue(['English', 'Spanish', 'Chinese', 'Dutch']);

      const { getByText } = render(<LanguageSelector />);
      
      // Open the modal
      const languageButton = getByText('üåê üá∫üá∏');
      fireEvent.press(languageButton);

      // Take snapshot with modal open
      const tree = render(<LanguageSelector />);
      // Manually set modal visible for snapshot
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot with modal open - Spanish current', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue('Spanish');
      mockTranslationHook.getAvailableLanguages.mockReturnValue(['English', 'Spanish', 'Chinese', 'Dutch']);

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot with limited language options', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue('English');
      mockTranslationHook.getAvailableLanguages.mockReturnValue(['English', 'Spanish']); // Only 2 languages

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot with single language option', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue('English');
      mockTranslationHook.getAvailableLanguages.mockReturnValue(['English']); // Only 1 language

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot with empty language options', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue('English');
      mockTranslationHook.getAvailableLanguages.mockReturnValue([]); // No languages

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Edge Cases', () => {
    it('should match snapshot when getCurrentLanguageDisplay returns null', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue(null);
      mockTranslationHook.getAvailableLanguages.mockReturnValue(['English', 'Spanish']);

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot when getCurrentLanguageDisplay returns undefined', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue(undefined);
      mockTranslationHook.getAvailableLanguages.mockReturnValue(['English', 'Spanish']);

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot when getAvailableLanguages returns null', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue('English');
      mockTranslationHook.getAvailableLanguages.mockReturnValue(null);

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot with very long language names', () => {
      mockTranslationHook.getCurrentLanguageDisplay.mockReturnValue('Very Long Language Name That Might Wrap');
      mockTranslationHook.getAvailableLanguages.mockReturnValue([
        'English',
        'Very Long Language Name That Might Wrap',
        'Another Extremely Long Language Name'
      ]);

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Props Variations', () => {
    it('should match snapshot with custom LANGUAGE_CODES', () => {
      const customTranslationHook = {
        ...mockTranslationHook,
        LANGUAGE_CODES: {
          'English': 'en-US',
          'Spanish': 'es-ES',
          'French': 'fr-FR',
          'German': 'de-DE'
        }
      };
      useTranslation.mockReturnValue(customTranslationHook);

      customTranslationHook.getCurrentLanguageDisplay.mockReturnValue('French');
      customTranslationHook.getAvailableLanguages.mockReturnValue(['English', 'Spanish', 'French', 'German']);

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot with missing LANGUAGE_CODES', () => {
      const customTranslationHook = {
        ...mockTranslationHook,
        LANGUAGE_CODES: {}
      };
      useTranslation.mockReturnValue(customTranslationHook);

      customTranslationHook.getCurrentLanguageDisplay.mockReturnValue('English');
      customTranslationHook.getAvailableLanguages.mockReturnValue(['English', 'Spanish']);

      const tree = render(<LanguageSelector />);
      expect(tree).toMatchSnapshot();
    });
  });
});