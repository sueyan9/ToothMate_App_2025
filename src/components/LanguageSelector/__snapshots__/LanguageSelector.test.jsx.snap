// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector Snapshot & Behavior Tests Behavior w/ language codes selecting Chinese calls changeLanguage with its CODE (zh) 1`] = `
[
  <TouchableOpacity
    onPress={[Function]}
    style={
      {
        "backgroundColor": "#f0f0f0",
        "borderRadius": 5,
        "marginHorizontal": 10,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      🌐 
      🇬🇧
    </Text>
  </TouchableOpacity>,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "maxWidth": 300,
          "padding": 20,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      >
        Select Language
      </Text>
      <TouchableOpacity
        onPress={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderBottomColor": "#eee",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "padding": 15,
            },
            {
              "backgroundColor": "#e3f2fd",
            },
          ]
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "marginRight": 12,
            }
          }
        >
          🇬🇧
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 16,
              },
              {
                "fontWeight": "bold",
              },
            ]
          }
        >
          English
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderBottomColor": "#eee",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "padding": 15,
            },
            false,
          ]
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "marginRight": 12,
            }
          }
        >
          🇪🇸
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 16,
              },
              false,
            ]
          }
        >
          Spanish
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderBottomColor": "#eee",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "padding": 15,
            },
            false,
          ]
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "marginRight": 12,
            }
          }
        >
          🇨🇳
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 16,
              },
              false,
            ]
          }
        >
          Chinese
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderBottomColor": "#eee",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "padding": 15,
            },
            false,
          ]
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "marginRight": 12,
            }
          }
        >
          🇳🇱
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 16,
              },
              false,
            ]
          }
        >
          Dutch
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={[Function]}
        style={
          {
            "backgroundColor": "#007AFF",
            "borderRadius": 5,
            "marginTop": 15,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Close
        </Text>
      </TouchableOpacity>
    </View>
  </View>,
]
`;

exports[`LanguageSelector Snapshot & Behavior Tests Closed Modal States should match snapshot with Chinese selected 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    🌐 
    🇨🇳
  </Text>
</TouchableOpacity>
`;

exports[`LanguageSelector Snapshot & Behavior Tests Closed Modal States should match snapshot with Dutch selected 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    🌐 
    🇳🇱
  </Text>
</TouchableOpacity>
`;

exports[`LanguageSelector Snapshot & Behavior Tests Closed Modal States should match snapshot with English selected 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    🌐 
    🌐
  </Text>
</TouchableOpacity>
`;

exports[`LanguageSelector Snapshot & Behavior Tests Closed Modal States should match snapshot with Spanish selected 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    🌐 
    🇪🇸
  </Text>
</TouchableOpacity>
`;

exports[`LanguageSelector Snapshot & Behavior Tests Edge Cases should match snapshot when getAvailableLanguages returns null 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    🌐 
    🇬🇧
  </Text>
</TouchableOpacity>
`;

exports[`LanguageSelector Snapshot & Behavior Tests Edge Cases should match snapshot when getCurrentLanguageDisplay returns null 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    🌐 
    🌐
  </Text>
</TouchableOpacity>
`;

exports[`LanguageSelector Snapshot & Behavior Tests Edge Cases should match snapshot when getCurrentLanguageDisplay returns undefined 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    🌐 
    🌐
  </Text>
</TouchableOpacity>
`;

exports[`LanguageSelector Snapshot & Behavior Tests Edge Cases should match snapshot with very long language names 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    🌐 
    🌐
  </Text>
</TouchableOpacity>
`;

exports[`LanguageSelector Snapshot & Behavior Tests Open Modal States should match snapshot with empty language options 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    🌐 
    🇬🇧
  </Text>
</TouchableOpacity>
`;

exports[`LanguageSelector Snapshot & Behavior Tests Open Modal States should match snapshot with limited language options 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    🌐 
    🇬🇧
  </Text>
</TouchableOpacity>
`;

exports[`LanguageSelector Snapshot & Behavior Tests Open Modal States should match snapshot with modal open - English current 1`] = `
[
  <TouchableOpacity
    onPress={[Function]}
    style={
      {
        "backgroundColor": "#f0f0f0",
        "borderRadius": 5,
        "marginHorizontal": 10,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      🌐 
      🇬🇧
    </Text>
  </TouchableOpacity>,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "maxWidth": 300,
          "padding": 20,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      >
        Select Language
      </Text>
      <TouchableOpacity
        onPress={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderBottomColor": "#eee",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "padding": 15,
            },
            {
              "backgroundColor": "#e3f2fd",
            },
          ]
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "marginRight": 12,
            }
          }
        >
          🇬🇧
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 16,
              },
              {
                "fontWeight": "bold",
              },
            ]
          }
        >
          English
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderBottomColor": "#eee",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "padding": 15,
            },
            false,
          ]
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "marginRight": 12,
            }
          }
        >
          🇪🇸
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 16,
              },
              false,
            ]
          }
        >
          Spanish
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderBottomColor": "#eee",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "padding": 15,
            },
            false,
          ]
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "marginRight": 12,
            }
          }
        >
          🇨🇳
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 16,
              },
              false,
            ]
          }
        >
          Chinese
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderBottomColor": "#eee",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "padding": 15,
            },
            false,
          ]
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "marginRight": 12,
            }
          }
        >
          🇳🇱
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 16,
              },
              false,
            ]
          }
        >
          Dutch
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={[Function]}
        style={
          {
            "backgroundColor": "#007AFF",
            "borderRadius": 5,
            "marginTop": 15,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Close
        </Text>
      </TouchableOpacity>
    </View>
  </View>,
]
`;

exports[`LanguageSelector Snapshot & Behavior Tests Open Modal States should match snapshot with single language option 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    🌐 
    🇬🇧
  </Text>
</TouchableOpacity>
`;

exports[`LanguageSelector Snapshot & Behavior Tests Props Variations should match snapshot with custom LANGUAGE_CODES 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "backgroundColor": "#f0f0f0",
      "borderRadius": 5,
      "marginHorizontal": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    🌐 
    🌐
  </Text>
</TouchableOpacity>
`;
